internal/rsl/rsl.go

```diff
@@ -487,8 +493,10 @@ func GetFirstEntry(repo *git.Repository) (*ReferenceEntry, []*AnnotationEntry, e
 					// The first entry cannot be an annotation
 					return nil, nil, ErrInvalidRSLEntry
 				}
-				firstEntry = entry
-				break
+				if entry.RefName == targetRef || targetRef == "" {
```

Comment:
I might have got it wrong last night. Does this still work when the first entry for a ref is actually the 17th entry overall? We hit this only when there aren't any parents, right? Do we want to continually track firstEntry outside this err block?

Reply:
I have fixed this, and changed the logic, the last check at the end is just if we never encounter a rsl entry